File: examples/js/agents/multi-agent.ts
/**
 * Multi-Agent Example
 * Demonstrates multiple agents working together
 */

import { Agent, PraisonAIAgents, Task } from 'praisonai';

async function main() {
  // Create specialized agents
  const researcher = new Agent({
    name: 'Researcher',
    role: 'Research Specialist',
    goal: 'Find accurate information on topics',
    instructions: 'You research topics thoroughly and provide factual information.'
  });

  const writer = new Agent({
    name: 'Writer',
    role: 'Content Writer',
    goal: 'Create engaging content from research',
    instructions: 'You write clear, engaging content based on provided research.'
  });

  // Create tasks
  const researchTask = new Task({
    description: 'Research the benefits of TypeScript over JavaScript',
    expectedOutput: 'A list of key benefits with explanations',
    agent: researcher
  });

  const writeTask = new Task({
    description: 'Write a short article about TypeScript benefits',
    expectedOutput: 'A 200-word article',
    agent: writer
  });

  // Run agents
  const agents = new PraisonAIAgents({
    agents: [researcher, writer],
    tasks: [researchTask, writeTask],
    process: 'sequential'
  });

  const result = await agents.start();
  console.log('Result:', result);
}

main().catch(console.error);

File: src/praisonai-rust/examples/multi_agent.rs
//! Multi-agent workflow example
//!
//! Run with: cargo run --example multi_agent

use praisonai::{Agent, AgentTeam, Process};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    // Create specialized agents
    let researcher = Agent::new()
        .name("researcher")
        .instructions("You are a research specialist. Find and summarize information.")
        .model("gpt-4o-mini")
        .build()?;
    
    let writer = Agent::new()
        .name("writer")
        .instructions("You are a skilled writer. Take research and create engaging content.")
        .model("gpt-4o-mini")
        .build()?;
    
    let editor = Agent::new()
        .name("editor")
        .instructions("You are an editor. Review and polish the content for clarity.")
        .model("gpt-4o-mini")
        .build()?;
    
    // Create a team with sequential process
    let team = AgentTeam::new()
        .agent(researcher)
        .agent(writer)
        .agent(editor)
        .process(Process::Sequential)
        .verbose(true)
        .build();
    
    println!("Team created with {} agents", team.len());
    println!("Process: Sequential (researcher -> writer -> editor)");
    
    // Run the workflow
    // Note: Requires OPENAI_API_KEY environment variable
    // let result = team.start("Write a blog post about Rust programming").await?;
    // println!("Final output:\n{}", result);
    
    println!("\nTo test with real API, set OPENAI_API_KEY and uncomment the start() call.");
    
    Ok(())
}
